shell/services/node/node_service.cc

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2022 Microsoft, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/services/node/node_service.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/command_line.h"
+#include "base/environment.h"
+#include "base/process/process.h"
+#include "base/strings/utf_string_conversions.h"
+#include "content/public/utility/utility_thread.h"
+#include "gin/data_object_builder.h"
+#include "gin/handle.h"
+#include "shell/browser/api/message_port.h"
+#include "shell/browser/javascript_environment.h"
+#include "shell/common/api/electron_bindings.h"
+#include "shell/common/gin_converters/callback_converter.h"
+#include "shell/common/gin_converters/file_path_converter.h"
+#include "shell/common/gin_helper/dictionary.h"
+#include "shell/common/gin_helper/event_emitter_caller.h"
+#include "shell/common/node_bindings.h"
+#include "shell/common/node_includes.h"
+#include "shell/common/node_util.h"
+#include "shell/common/v8_value_serializer.h"
+
+namespace electron {
+
+NodeService::NodeService(
+    mojo::PendingReceiver<node::mojom::NodeService> receiver)
+    : node_bindings_(
+          NodeBindings::Create(NodeBindings::BrowserEnvironment::kUtility)),
+      electron_bindings_(
+          std::make_unique<ElectronBindings>(node_bindings_->uv_loop())) {
+  if (receiver.is_valid())
+    receiver_.Bind(std::move(receiver));
+  if (auto* thread = content::UtilityThread::Get()) {
```

Comment:
when is this nullptr? should we CHECK if it is?

Reply:
This should never be `nullptr` in our use, only when the call is not made from `UtilityThread`(main thread of the utility process) this will be `nullptr` https://source.chromium.org/chromium/chromium/src/+/main:content/utility/utility_main.cc;l=207-208. I can make this into a CHECK

Reply:
https://github.com/electron/electron/pull/34980/commits/4a3a57985a314f72bf2afa0b8280d812aeec9c6a